python/tests/test_stats.py

```diff
@@ -381,6 +381,123 @@ def naive_genealogical_nearest_neighbours(ts, focal, reference_sets):
     return A
 
 
+def parse_time_windows(ts, time_windows):
+    if time_windows is None:
+        time_windows = [0.0, ts.max_root_time]
+    return np.array(time_windows)
+
+
+def windowed_genealogical_nearest_neighbours(
+    ts,
+    focal,
+    reference_sets,
+    windows=None,
+    time_windows=None,
+    span_normalise=True,
+    time_normalise=True,
+):
+    """
+    genealogical_nearest_neighbours with support for span- and time-based windows
+    """
+    reference_set_map = np.zeros(ts.num_nodes, dtype=int) - 1
+    for k, reference_set in enumerate(reference_sets):
+        for u in reference_set:
+            if reference_set_map[u] != -1:
+                raise ValueError("Duplicate value in reference sets")
+            reference_set_map[u] = k
+
+    windows_parsed = ts.parse_windows(windows)
```

Comment:
It's a bit simpler if you just say
```
windows = ts.parse_windows(windows)
```
there's no real reason to keep the original value lying around, and it's a potential source of errors.

Reply:
I kept the original around so I could reshape the output on lines 477-487. I'll rewrite to store a boolean if `windows` and `time_windows` are used.

Reply:
Ah, I see. It's probably good idea to store the ``output_dims`` explicitly at the top then, so it's clear what you're deriving this from.